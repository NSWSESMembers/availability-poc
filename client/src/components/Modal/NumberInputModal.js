import React from 'react';

import PropTypes from 'prop-types';
import { View, Modal, TouchableOpacity, TouchableWithoutFeedback, Text, TextInput } from 'react-native';

import styles from './styles';

const NumberInputModal = ({
  title,
  placeHolder,
  closeModal,
  backModal,
  onChangeText,
  visible,
  onSave,
}) => (
  <View>
    <Modal
      transparent
      visible={visible}
      onRequestClose={closeModal}
      animationType="fade"
    >
      <TouchableWithoutFeedback
        onPress={() => closeModal}
      >
        <View style={styles.overlayStyle}>
          <View style={styles.headerContainer}>
            <TouchableOpacity>
              <View style={styles.headerStyle}>
                <Text style={styles.headerTextStyle}>{title}</Text>
              </View>
            </TouchableOpacity>
          </View>
          <View style={styles.optionContainer}>
            <View>
              <View style={{ paddingHorizontal: 10 }}>
                <TextInput
                  style={styles.etaInput}
                  keyboardType="numeric"
                  placeholder={placeHolder}
                  maxLength={10}
                  onChangeText={onChangeText}
                />
              </View>
            </View>
          </View>
          <View style={styles.nextContainer}>
            <TouchableOpacity onPress={() => onSave}>
              <View style={styles.nextStyle}>
                <Text style={styles.nextTextStyle}>Next</Text>
              </View>
            </TouchableOpacity>
          </View>
          <View style={styles.cancelContainer}>
            <TouchableOpacity onPress={() => backModal}>
              <View style={styles.cancelStyle}>
                <Text style={styles.cancelTextStyle}>Back</Text>
              </View>
            </TouchableOpacity>
          </View>
        </View>
      </TouchableWithoutFeedback>
    </Modal>
  </View>
);

NumberInputModal.propTypes = {
  title: PropTypes.string.isRequired,
  placeHolder: PropTypes.string.isRequired,
  closeModal: PropTypes.func.isRequired,
  onSave: PropTypes.func.isRequired,
  backModal: PropTypes.func.isRequired,
  onChangeText: PropTypes.func.isRequired,
  visible: PropTypes.bool.isRequired,
};

export default NumberInputModal;
