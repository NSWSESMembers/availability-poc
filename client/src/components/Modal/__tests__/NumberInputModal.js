import React from 'react';
import renderer from 'react-test-renderer';

import NumberInputModal from '../NumberInputModal';

const noop = () => {};

test('renders correctly', () => {
  const tree1 = renderer.create(
    <NumberInputModal
      title="title"
      placeHolder="placeHolder"
      closeModal={noop}
      onSave={noop}
      backModal={noop}
      onChangeText={noop}
      visible
    />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();
});
