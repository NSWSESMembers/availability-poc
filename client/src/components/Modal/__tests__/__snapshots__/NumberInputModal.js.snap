// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.7)",
          "flex": 1,
          "justifyContent": "center",
          "padding": "5%",
        }
      }
      testID={undefined}
    >
      <View
        style={
          Object {
            "alignSelf": "stretch",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(255,255,255,0.8)",
                "borderRadius": 5,
                "padding": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#333",
                  "fontSize": 16,
                  "textAlign": "center",
                }
              }
            >
              title
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.8)",
            "borderRadius": 5,
            "flexShrink": 1,
            "marginBottom": 8,
            "marginTop": 8,
            "padding": 8,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "paddingHorizontal": 10,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              keyboardType="numeric"
              maxLength={10}
              onChangeText={[Function]}
              placeholder="placeHolder"
              style={
                Object {
                  "color": "red",
                  "fontSize": 20,
                  "height": 100,
                  "textAlign": "center",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "stretch",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(255,255,255,0.8)",
                "borderRadius": 5,
                "marginBottom": 8,
                "padding": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "green",
                  "fontSize": 16,
                  "textAlign": "center",
                }
              }
            >
              Next
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "stretch",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(255,255,255,0.8)",
                "borderRadius": 5,
                "padding": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#333",
                  "fontSize": 16,
                  "textAlign": "center",
                }
              }
            >
              Back
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
